src/components/DisplayInput.js

```diff
@@ -54,23 +55,31 @@ const DisplayInput = React.createClass({
         <div style={styles.wrapper}>
           <Row>
             {this.props.label ? (
-              <Column span={labelColumn} style={styles.label}>
-                <div key='label' style={styles.label}>
+              <Column span={labelColumn}>
+                <div>
                   <div style={Object.assign({}, styles.labelText, this.props.labelStyle)}>
                     {this.props.label}
                   </div>
                 </div>
               </Column>
             ) : null }
 
-            <Column span={inputColumn}>
-              <input
-                {...this.props}
-                key='input'
-                label={this.props.label}
-                style={styles.input}
-                type='text'
-              />
+            <Column relative={!hasChildren} span={inputColumn}>
+              {this.props.children ? (
+                <div style={styles.children}>
+                  {this.props.children}
+                </div>
+              ) : (
+                <div style={styles.inputWrapper}>
+                  <input
+                    {...this.props}
```

Comment:
Yeah I didn't go down that rabbit hole on purpose for this PR but I agree we need to do some clean up around stuff like that.